src/main/java/uk/gov/dbt/ndtp/federator/jobs/handlers/ClientDynamicConfigJob.java

```
@@ -0,0 +1,258 @@
+
+    private static FederatorConfigurationService staticService;
+    private final FederatorConfigurationService configService;
+    private final JobSchedulerProvider scheduler;
+
+    /** Creates a new job using static service reference. Used by JobRunr when deserializing jobs. */
+    public ClientDynamicConfigJob() {
+        this.configService = staticService;
+        this.scheduler = DefaultJobSchedulerProvider.getInstance();
+    }
+
+    /**
+     * Creates a new job with specified dependencies.
+     *
+     * @param configService the configuration service
+     * @param scheduler the job scheduler provider
+     */
+    public ClientDynamicConfigJob(
+            final FederatorConfigurationService configService, final JobSchedulerProvider scheduler) {
+        this.configService = configService;
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Initializes the static service reference. Must be called before JobRunr executes jobs.
+     *
+     * @param service the configuration service
+     */
+    public static void initialize(final FederatorConfigurationService service) {
+        staticService = service;
+        log.debug("ClientDynamicConfigJob initialized");
+    }
+
+    /**
+     * Executes the configuration update job.
+     *
+     * @param value the job parameters containing node ID
+     */
+    @Override
+    public void run(final JobParams value) {
+        final String nodeId = resolveNodeId(value);
+        log.info(LOG_START, nodeId);
+
+        try {
+            final ProducerConfigDTO config = configService.getProducerConfiguration();
+            if (config == null) {
+                log.warn(LOG_NO_CONFIG, nodeId);
+                return;
+            }
+            reloadJobs(config, nodeId);
+        } catch (ManagementNodeDataException e) {
+            log.error(LOG_ERROR, nodeId, e.getMessage(), e);
+        } catch (Exception e) {
+            log.error(LOG_ERROR_UNEXP, nodeId, e.getMessage(), e);
```

We are swallowing exception , might be good idea to throw some Federator exception which extends Runtime exception

---

we are not swallowing, we are logging them. and even showing their stacktrace in the logs? am I missing something here?